edx_solutions_api_integration/management/commands/add_test_poll_responses.py

```diff
@@ -0,0 +1,198 @@
+from xmodule.modulestore.django import modulestore
+
+
+# Variables
+
+logger = logging.getLogger(__name__)  # pylint: disable=locally-disabled, invalid-name
+
+# Classes
+
+
+class Command(BaseCommand):
+    """
+    Command to generate dummy problem responses xblock poll in a course
+    """
+    help = 'Generates dummy responses for users enrolled in a course'
+    option_list = BaseCommand.option_list + (
+        make_option(
+            "--course-id",
+            dest="course_id",
+            help="Course ID",
+        ),
+        make_option(
+            "--num-responses",
+            dest="num_responses",
+            default=10000,
+            type="int",
+            help="Limit generation of responses to this number. Defaults to "
+                 "generating 10k responses. Setting this to 0 will generate"
+                 "a dummy response for every user in every poll.",
+        ),
+        make_option(
+            "--batch-size",
+            dest="batch_size",
+            default=500,
+            type="int",
+            help="Batch size to use when adding responses to the database.",
+        ),
+    )
+    state_summary = {}
+
+    @staticmethod
+    def get_course_blocks(  # pylint: disable=bad-continuation
+        store,
+        course_key,
+        categories,
+        revision=ModuleStoreEnum.RevisionOption.published_only
+    ):
+        """
+        Retrieve all XBlocks in the course for a particular category.
+
+        By default, only returns only XBlocks that are published.
+        """
+        return [
+            block for block in store.get_items(
+                course_key,
+                qualifiers={"category": categories},
+                revision=revision,
+            )
+        ]
+
+    @staticmethod
+    def get_enrolled_students(course_key):
+        """
+        Given a course_id, returns a QuerySet of all the students
+        in the course.
+        """
+        return User.objects.filter(courseenrollment__course_id=course_key)
+
+    @staticmethod
+    def generate_combinations(students, blocks, limit):
+        """
+        Generates all possible user X block combination that aren't
+        already on the database
+        """
+        # User Cartesian multiplication to find all possible
+        # student vs block combinations.
+        combinations = product(students, blocks)
+
+        # Remove combinations that already have entries on the answers table
+        responses = StudentModule.objects.filter(
+            student__in=students,
+            module_state_key__in=[block.location for block in blocks]
+        )
+        # Compile list of combinations to be skipped and map course run info.
+        # The casting to str() is needed to avoid the need to call
+        # map_to_course on every location, which could lead to a query explosion
+        skip_combinations = [
+            [item.student, str(item.module_state_key)] for item in responses
```

Comment:
Make this a set of tuples instead of a list of lists. That way lookups are instant. 

i.e.
```python
skip_cominations = {(item.student, str(item.module_state_key)) for item in responses}
```

Reply:
Done.

Reply:
Done!